pyrokinetics/gene.py

```diff
@@ -0,0 +1,370 @@
+import f90nml
+import copy
+from .constants import *
+from .local_species import LocalSpecies
+from .numerics import Numerics
+from .gk_code import GKCode
+import os
+from path import Path
+from cleverdict import CleverDict
+
+class GENE(GKCode):
+    """
+    Basic GENE object
+
+    """
+
+    def __init__(self):
+
+        self.base_template_file = os.path.join(Path(__file__).dirname(), 'templates', 'input.gene')
+        self.default_file_name = 'input.gene'
+
+    def read(self, pyro, data_file=None, template=False):
+        """
+        Reads GENE input file into a dictionary
+        """
+
+        if template and data_file is None:
+            data_file = self.base_template_file
+
+        gene = f90nml.read(data_file).todict()
+
+        pyro.initial_datafile = copy.copy(gene)
+
+        try: 
+            nl_flag = gene['general']['nonlinear']
+        
+            if nl_flag == '.F.':
+
+                pyro.linear = True
+            else: 
+                pyro.linear = False 
+        except KeyError:
+            pyro.linear = True 
+
+
+
+        pyro.gene_input = gene
+
+ #       # Loads pyro object with equilibrium data
+        if not template:
+            self.load_pyro(pyro)
+
+        # Load Pyro with numerics if they don't exist
+        if not hasattr(pyro, 'numerics'):
+            self.load_numerics(pyro, gene)
+
+    def load_pyro(self, pyro):
+        """
+        Loads GENE input into Pyro object
+        """
+
+        # Geometry
+        gene = pyro.gene_input
+
+        gene_eq = gene['geometry']['magn_geometry']
+
+        if gene_eq == 's_alpha': 
+            pyro.local_geometry = 'SAlpha'
+        elif gene_eq == 'miller': 
+            pyro.local_geometry = 'Miller'
+
+
+        #  Load GENE with local geometry
+        self.load_local_geometry(pyro, gene)
+
+        # Load GENE with species data
+        self.load_local_species(pyro, gene)
+
+
+ # Need species to set up beta_prime
+
+        if pyro.local_geometry_type == 'Miller':
+            if pyro.local_geometry['Bunit'] is not None:
+                pyro.local_geometry['beta_prime'] = - pyro.local_species['a_lp'] / pyro.local_geometry['Bunit'] ** 2
```

Comment:
@dake0795 I this needs to be 

```suggestion
            if pyro.local_geometry['B0'] is not None:
                pyro.local_geometry['beta_prime'] = - pyro.local_species['a_lp'] / pyro.local_geometry['B0'] ** 2
```